### ISSUE TITLE
Bug Report: `unicorn/no-useless-undefined` incorrectly flags valid argument

### ISSUE THREAD

**Issue by y-hsgw** [sindresorhus/eslint-plugin-unicorn#2616]
The `unicorn/no-useless-undefined` rule incorrectly reports an error when explicitly passing undefined as an argument, even when the function is defined to accept undefined explicitly (e.g., string | undefined in TypeScript). This is a valid and meaningful usage that shouldn't be removed.

Rule: `unicorn/no-useless-undefined`

**❌ Incorrectly flagged:**
```js
const fn = (input: string | undefined) => {
	console.log(input);
};

fn(undefined); // <-- incorrectly reported
```

**✅ Should be valid:**
This is a meaningful use of undefined because the function is explicitly typed to accept it. Removing the undefined would change the function call semantics.

**Comment by fisker** [comment]
Duplicate of #2556
